virtcontainers/device/drivers/vhost_user_blk.go

```diff
@@ -35,30 +43,75 @@ func (device *VhostUserBlkDevice) Attach(devReceiver api.DeviceReceiver) (err er
 	if skip {
 		return nil
 	}
+
+	// Since vhost-user-blk uses "vd" prefix in Linux kernel, not "sd",
+	// sandbox block index should be updated only if sandbox default block
+	// driver is "virtio-blk"/"virtio-blk-ccw"/"virtio-mmio" which uses
+	// "vd" prefix in Linux kernel.
+	updateBlockIndex := false
+	customOptions := device.DeviceInfo.DriverOptions
+	if customOptions["block-driver"] == "virtio-blk" ||
+		customOptions["block-driver"] == "virtio-blk-ccw" ||
+		customOptions["block-driver"] == "virtio-mmio" {
+		updateBlockIndex = true
+	}
+
+	if updateBlockIndex {
+		_, err = devReceiver.GetAndSetSandboxBlockIndex()
+	}
+
 	defer func() {
 		if err != nil {
+			if updateBlockIndex {
+				devReceiver.DecrementSandboxBlockIndex()
+			}
 			device.bumpAttachCount(false)
 		}
 	}()
 
-	// generate a unique ID to be used for hypervisor commandline fields
-	randBytes, err := utils.GenerateRandomBytes(8)
 	if err != nil {
 		return err
 	}
-	id := hex.EncodeToString(randBytes)
 
-	device.DevID = id
-	device.Type = device.DeviceType()
+	vAttrs := &config.VhostUserDeviceAttrs{
+		DevID:      utils.MakeNameID("blk", device.DeviceInfo.ID, maxDevIDSize),
+		SocketPath: device.DeviceInfo.HostPath,
+		Type:       config.VhostUserBlk,
+	}
 
-	return devReceiver.AppendDevice(device)
+	deviceLogger().WithField("device", device.DeviceInfo.HostPath).WithField("SocketPath", vAttrs.SocketPath).Infof("Attaching %s device", config.VhostUserBlk)
```

Comment:
please, don't use `Infof` and try with `WithFields`

Reply:
Ack. Using `Info` with `WithField`.

...Actually, it referenced `attach` function in file `block.go`